src/crewai/agent.py

```diff
@@ -264,15 +267,63 @@ def execute_task(
                     task=task,
                 ),
             )
-            result = self.agent_executor.invoke(
-                {
-                    "input": task_prompt,
-                    "tool_names": self.agent_executor.tools_names,
-                    "tools": self.agent_executor.tools_description,
-                    "ask_for_human_input": task.human_input,
-                }
-            )["output"]
+
+            if not isinstance(self.max_execution_time, int) or self.max_execution_time <= 0:
+                raise ValueError("Max Execution time must be a positive integer greater than zero")
+            
+            # Apply timeout if specified
+            if self.max_execution_time is not None:
+                import concurrent.futures
+                with concurrent.futures.ThreadPoolExecutor() as executor:
+                    future = executor.submit(
+                        self.agent_executor.invoke,
+                        {
+                            "input": task_prompt,
+                            "tool_names": self.agent_executor.tools_names,
+                            "tools": self.agent_executor.tools_description,
+                            "ask_for_human_input": task.human_input,
+                        }
+                    )
+                    
+                    try:
+                        result = future.result(timeout=self.max_execution_time)["output"]
```

Comment:
what about creating a two couple of function to process with timeout and without timeout? Just to try split this code that has a insane identation haha